### ISSUE TITLE
`WRONG_INDENTATION`: false positives for parentheses-surrounded infix expressions

### ISSUE THREAD

**Issue by 0x6675636b796f75676974687562** [saveourtool/diktat#1409]
The following code fragment triggers no errors, provided that `expectedIndentAfterOperators` is **off**:

```kotlin
val value = 1 to
    2
```

Now, let's surround the expression (which is an infix function call) with parentheses, preserving the indent:

```kotlin
val value = (1 to
    2)
```

_DiKTat_ will report the following errors:

```
[WRONG_INDENTATION] only spaces are allowed for indentation and each indentation should equal to 4 spaces (tabs are not allowed): expected 8 but was 4
[WRONG_INDENTATION] only spaces are allowed for indentation and each indentation should equal to 4 spaces (tabs are not allowed): expected 8 but was 4
```

Strangely, if we move the expression onto the next line (and increment the indent), the errors immediately go away:

```kotlin
val value =
    (1 to
        2)
```

## Environment information
* diktat version: 1.2.0
* build tool (maven/gradle): Maven
* how is diktat run (CLI, plugin, etc.): Maven plug-in
* kotlin version: 1.7.0
* operating system: Windows

**Comment by 0x6675636b796f75676974687562** [comment]
If we take a test for `ExpressionIndentationChecker` ([`IndentationRuleWarnTest.closing parenthesis bug()`](https://github.com/saveourtool/diktat/blob/master/diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/spaces/IndentationRuleWarnTest.kt#L659)) and reformat the original code:

```kotlin
fun foo() {
    return x +
        (y +
            foo(x)
        )
}
```

&mdash; and reformat it: 

```kotlin
fun foo() {
    return x + (y +
        foo(x)
    )
}
```

&mdash; the test will still pass. If, however, we convert the `return` statement to an expression body, it will immediately start to fail:

```kotlin
fun foo() = x + (y +
    foo(x)
)
```
